src/DssDirectDepositAaveDai.sol

```diff
@@ -381,20 +467,21 @@ contract DssDirectDepositAaveDai {
     function cull() external {
         require(vat.live() == 1, "DssDirectDepositAaveDai/no-cull-during-shutdown");
         require(live == 0, "DssDirectDepositAaveDai/live");
-        require(_add(tic, tau) <= block.timestamp, "DssDirectDepositAaveDai/early-cull");
+        require(_add(tic, tau) <= block.timestamp || wards[msg.sender] == 1, "DssDirectDepositAaveDai/early-cull");
         require(culled == 0, "DssDirectDepositAaveDai/already-culled");
 
         (uint256 ink, uint256 art) = vat.urns(ilk, address(this));
         require(ink <= 2 ** 255, "DssDirectDepositAaveDai/overflow");
         require(art <= 2 ** 255, "DssDirectDepositAaveDai/overflow");
-        vat.grab(ilk, address(this), address(this), address(chainlog.getAddress("MCD_VOW")), -int256(ink), -int256(art));
+        vat.grab(ilk, address(this), address(this), chainlog.getAddress("MCD_VOW"), -int256(ink), -int256(art));
```

Comment:
I would almost rather the `vow` be fileable than rely on the chainlog for calls like this.

Using the chainlog here requires that it becomes a test dependency, it just seems cleaner to use a storage slot for this.

Reply:
for my learning: did you remove `address` because it would already be returned as one and no need to cast?

Reply:
Yeah the return is already an address. See: https://github.com/makerdao/dss-direct-deposit/pull/10/files#R30

Reply:
We've discussed this before. I really think it's a mistake to have to synchronously update the `vow` in every module if/when we update it. Having a dynamic lookup ensures we don't miss anything on the periphery.

Reply:
Yeah I'm still dubious about it as for example in `dss-deploy-scripts` I had a problem with relying the changelog to deploy contracts. The changelog will be something that we will probably update after redeploying (like we did in Goerli). As if the modules requires the chainlog to load addresses we are in a problem for that case.

Reply:
We will keep discussing this as I think the usage of the chainlog for the `end` is even more controversial.
But that can be addressed in a different PR.